jsk_pcl_ros_utils/src/polygon_magnifier_nodelet.cpp

```diff
@@ -73,12 +73,26 @@ namespace jsk_pcl_ros_utils
     boost::mutex::scoped_lock lock(mutex_);
     vital_checker_->poke();
 
-    jsk_recognition_msgs::PolygonArray ret_polygon_array = *msg;
+    jsk_recognition_msgs::PolygonArray ret_polygon_array;
+    ret_polygon_array.header = msg->header;
 
     for (size_t i = 0; i < msg->polygons.size(); i++) {
-      jsk_recognition_utils::ConvexPolygon poly = jsk_recognition_utils::ConvexPolygon::fromROSMsg(msg->polygons[i].polygon);
-      jsk_recognition_utils::ConvexPolygon::Ptr magnified_poly = poly.magnifyByDistance(magnify_distance_);
-      ret_polygon_array.polygons[i].polygon = magnified_poly->toROSMsg();
+      jsk_recognition_utils::ConvexPolygon poly =
+        jsk_recognition_utils::ConvexPolygon::fromROSMsg(msg->polygons[i].polygon);
+
+      jsk_recognition_utils::ConvexPolygon::Ptr magnified_poly
+        = poly.magnifyByDistance(magnify_distance_);
+
+      if (!magnified_poly->isConvex()) {
+        ROS_ERROR("Magnified polygon %ld is not convex.", i);
```

Comment:
I think it is OK to push magnified polygon even if it is non-convex. In some cases, we want the result even if it is non-convex. Keeping length of polygon array is easy to manage.
How about following (just add warn message if non-convex)?
```
    jsk_recognition_msgs::PolygonArray ret_polygon_array = *msg;

    for (size_t i = 0; i < msg->polygons.size(); i++) {
      jsk_recognition_utils::ConvexPolygon poly = jsk_recognition_utils::ConvexPolygon::fromROSMsg(msg->polygons[i].polygon);
      jsk_recognition_utils::ConvexPolygon::Ptr magnified_poly = poly.magnifyByDistance(magnify_distance_);
      if (!magnified_poly->isConvex()) {
              ROS_WARN("Magnified polygon %ld is not convex.", i);
      }

      ret_polygon_array.polygons[i].polygon = magnified_poly->toROSMsg();
    }
    pub_.publish(ret_polygon_array);
```